php/Block.php

```diff
@@ -63,66 +63,179 @@ public function register_block() {
 	 * @return string The markup of the block.
 	 */
 	public function render_callback( $attributes, $content, $block ) {
-		$post_types = get_post_types(  [ 'public' => true ] );
-		$class_name = $attributes['className'];
-		ob_start();
 
+		// Store this in a local var for caching since we'll reference it a few
+		// times in this method.
+		$current_post_id = get_the_ID();
+
+		/**
+		 * Fetch all public post types as objects, and construct a new array that
+		 * maps the post type's label name (not slug), and the number of
+		 * published posts within that post type so we can easily display it
+		 * below.
+		 */
+
+		// Array to store our new values in.
+		$public_post_type_counts = [];
+
+		// Get all public post type objects.
+		$post_type_objects = get_post_types(  [ 'public' => true ], 'objects' );
+
+		// Validate the values.
+		if (
+			! empty( $post_type_objects )
+			&& is_array( $post_type_objects )
```

Comment:
Specific project or company code standards may prefer the operator at the end of the preceding line, but I find it more readable this way.